content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,165 @@
+
+## Explicit saving
+An explicit save does not apply changes until the user explicitly takes an action to submit the changes.
+
+Explicitly saved forms can be saved by clicking a submit button or by pressing the "Enter" key while focused on an input.
+
+If you're designing a form, start with an explicit saving pattern. Avoid mixing explicit and automatic save patterns on a single page with multiple forms, and never mix save patterns in a single form. For example: if I upload a profile photo in the "Public profile" form, that change should not be saved until I explicitly submit the whole form.
+
+If there was an error saving the data, the user's data should be preserved in the form and they should be given [feedback](#feedback) about the failure.
+
+If a user tries to navigate away from a page with a form that has unsaved changes, you can optionally warn them that their changes will be lost using the [beforeunload event](https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event). Modern browsers prevent developers from customizing the alrt message, so we're limited to a generic message like "Do you want to leave this site? Changes you made may not be saved."
+
+### Declarative inputs
+
+When designing a form that is saved explicitly, use the following inputs:
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Browser-native `<select>` dropdown menus
+- Custom dropdown menus that allow multiple selections
+
+To maintain consistent interaction patterns across your app, do not use toggle switches or segmented controls in an explicitly saved form. They should only be used when they automatically change based on user input.
+
+To build an accessible and consistent app, do not automatically save declarative inputs.
+
+### Problems with autosaving:
+**Text inputs**
+- Users are expecting to set a value and then submit
+- Sensitive information could be unintentionally submitted. For example: a change password input
+
+**Checkbox groups**
+- It could be unclear whether or not their selections have been saved
+- Users may accidentally submit a selection that they clicked by accident
+
+**Radio button groups**
+Screenreader users can't read the options without selecting them. This could accidentally apply a selection the user didn't want.
+
+**Browser-native `<select>` dropdowns**
+A select dropdown has a similar problem as a radio button group. Options are selected when a user focuses the select and uses their arrow keys to read through each option
+
+## Automatic saving
+
+Automatic saving applies changes as they are made.
+
+Automatic saving should be used when the user expects instant feedback from the change they made. Changes should only be saved automatically using [imperitive inputs](#imperitive-inputs) and for individual settings that are not changed via separate flows.
```

Comment:
Do you mean imperative? If so, find a replace in the entire doc 😬

Reply:
> individual settings that are not changed via separate flows.

I don't understand what this means.

Reply:
It means edits that don't bring you into a new page or modal dialog. For example: adding a milestone to an issue.

I'm just removing that part because I think the first part of the sentence covers it.